solarwinds_apm/api/__init__.py

```diff
@@ -41,6 +43,13 @@ def set_transaction_name(custom_name: str) -> bool:
      from solarwinds_apm.api import set_transaction_name
      result = set_transaction_name("my-foo-name")
     """
+    if isinstance(get_tracer_provider(), NoOpTracerProvider):
+        logger.debug(
+            "Cannot cache custom transaction name %s because agent not enabled; ignoring",
+            custom_name,
+        )
+        return False
```

Comment:
I don't think we have documented recommendations for return values for agents in no-op mode. For the AppOptics java agent and the .net library when in no-op methods return `True` except for methods like `IsAgentAvailable`. 

@cheempz should there be a standardized behavior for no-op methods?

Reply:
Good point, I guess the rationale for AO Java/.NET returning `True` in almost all cases is to minimize the chance that in no-op mode our API calls might cause unnecessary handling of error response in the customer's application code making these calls.

It looks like SWO Java is also returning False when in no-op mode https://github.com/appoptics/solarwinds-apm-java/blob/main/solarwinds-otel-sdk/src/main/java/com/appoptics/api/ext/SolarWindsAgent.java#L37 so we have mini-standardization between the custom distros ;) But given these are our own API we can get them to all agree -- I do lean towards returning True across languages in this case. WDYT @tammy-baylis-swi @CleverChuk?

Reply:
> minimize the chance that in no-op mode our API might cause unnecessary handling of error response in the customer's application code making these calls

This makes sense to me! 👍 `set_transaction_name` now returns True for any case in [3d503c9](https://github.com/solarwindscloud/solarwinds-apm-python/pull/126/commits/3d503c90973efd0df7869341f47cc5c71711de3b)

Reply:
Correction in [fbf687b](https://github.com/solarwindscloud/solarwinds-apm-python/pull/126/commits/fbf687bb55032977e1fba2bc68542dd6c2ad3b45)!:

[Agent not enabled](https://github.com/solarwindscloud/solarwinds-apm-python/blob/fbf687bb55032977e1fba2bc68542dd6c2ad3b45/solarwinds_apm/api/__init__.py#L51) --> True
[Processor not configured](https://github.com/solarwindscloud/solarwinds-apm-python/blob/fbf687bb55032977e1fba2bc68542dd6c2ad3b45/solarwinds_apm/api/__init__.py#L66) --> False
[Span not started](https://github.com/solarwindscloud/solarwinds-apm-python/blob/fbf687bb55032977e1fba2bc68542dd6c2ad3b45/solarwinds_apm/api/__init__.py#L75) --> False
[Able to cache custom name](https://github.com/solarwindscloud/solarwinds-apm-python/blob/fbf687bb55032977e1fba2bc68542dd6c2ad3b45/solarwinds_apm/api/__init__.py#L83) --> True